source/extensions/filters/http/aws_request_signing/config.cc

```diff
@@ -51,27 +52,45 @@ AwsRequestSigningFilterFactory::createFilterFactoryFromProtoTyped(
     const AwsRequestSigningProtoConfig& config, const std::string& stats_prefix, DualInfo dual_info,
     Server::Configuration::ServerFactoryContext& server_context) {
 
+  std::string region;
+  region = config.region();
+
+  if (region.empty()) {
+    auto region_provider = std::make_shared<Extensions::Common::Aws::RegionProviderChain>();
+    absl::optional<std::string> regionopt = region_provider->getRegion();
+    if (!regionopt.has_value()) {
+      throw EnvoyException("Region string cannot be retrieved from configuration, environment or "
+                           "profile/config files.");
```

Comment:
Do you want to consider changing the exception message to something like below ?
`"AWS region is not set in xDS configuration and failed to retrieve from environment variable or AWS profile/config files."`

Reply:
Similar exception message inside perRoute function

Reply:
done